arch-snapshot/arch.md

```diff
@@ -0,0 +1,2921 @@
+# Golem — current architecture
```

Comment:
My main problem with this document is that it is sadly a tedious, uninspiring read. 

An architecture document should convey the most interesting and novel ideas as well and strive to inspire. It may be technically challenging but this should be balanced by a convinvcing story and gripping storytelling. Alas, this document fails to achieve either of this goals.